runtime/runtime/src/tests/apply.rs

```diff
@@ -86,20 +119,22 @@ fn setup_runtime_for_shard(
     let runtime = Runtime::new();
     let mut signers = vec![];
     let mut initial_state = tries.new_trie_update(shard_uid, root);
-    for account_id in initial_accounts.into_iter() {
-        let signer: Arc<Signer> = Arc::new(InMemorySigner::test_signer(&account_id));
+    for (account_id, signers_for_account) in accounts_with_keys.into_iter() {
         let mut initial_account = account_new(initial_balance, CryptoHash::default());
         // For the account and a full access key
         initial_account.set_storage_usage(182);
         initial_account.set_locked(initial_locked);
+
         set_account(&mut initial_state, account_id.clone(), &initial_account);
-        set_access_key(
-            &mut initial_state,
-            account_id,
-            signer.public_key(),
-            &AccessKey::full_access(),
-        );
-        signers.push(signer);
+        for signer in signers_for_account {
+            set_access_key(
+                &mut initial_state,
+                account_id.clone(),
+                signer.public_key(),
+                &AccessKey::full_access(),
+            );
+            signers.push(signer);
+        }
```

Comment:
you can do something like `signers_for_account.into_iter().map(...).collect();`. Then you do not need to define signers as `mut` above and have to push to it.